### PR TITLE
Use submodule to import vision and vision android repo.

### LINKED ISSUE #111: Use submodule to import development repos
Our current way to config development repos is not very convince to config locally, an alternative solution is to use submodules importing these development repos. So that we will only need one param `BUILD_CORE_FROM_SOURCE` to config the project choosing maven or codes for compiling.

### PR DESCRIPTION
**Pull request by Chaoba**
Resolves #111

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
-    project(':VisionAr').projectDir = new File(VISION_AR)
+File secretPropsFile = file('./local.properties')
+if (secretPropsFile.exists()) {
+    Properties p = new Properties()
+    p.load(new FileInputStream(secretPropsFile))
+    p.each { name, value ->
+        gradle.ext.set name, value
+    }
+} else {
+    throw new IllegalStateException("secret.properties could not be located for build process")
```
Comment (yunikkk) [inline settings.gradle]: local.properties

**Comment by korshaknn** [comment]
@yunikkk maybe we should hold on current PR until a PR  with a new modules logic is merged?

**Comment by yunikkk** [comment]
LGTM, guys WDYT?

**Comment by yunikkk** [comment]
Thanks @Chaoba! And sorry for the long review process =)

**Pull request merged**